src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -2973,6 +2979,30 @@ private string GetAssemblyPathInGac(AssemblyNameExtension assemblyName, SystemPr
         /// <returns>True if there was success.</returns>
         public override bool Execute()
         {
+            if(UseResolveAssemblyReferenceService && BuildEngine is IRarBuildEngine rarBuildEngine)
+            {
+                using var client = new RarClient(rarBuildEngine);
+
+                var connected = client.Connect();
+                if (!connected)
+                {
+                    if(client.CreateNode())
+                    {
+                        connected = client.Connect();
+                    }
+                }
+
+                if(connected)
+                {
+                    // Client is connected to the RAR node, we can execute RAR task remotely
+                    // return client.Execute(); // TODO: Let it do something.
+                    var number = client.GetNumber(42);
```

Comment:
is 42 a magic number?

Reply:
I have it here just for testing purpose. Will delete that when converting it into final PR.